mesos-docker/run/run.sh

```diff
@@ -122,11 +122,37 @@ function check_if_container_is_up {
   fi
 }
 
+function quote_if_non_empty {
+
+  if [[ -n $1 ]];then
+    echo "\"$1\""
+  else
+    echo ""
+  fi
+
+}
+
+function check_mesos_version {
+
+  MIT_MESOS_HOST_VERSION=$(dpkg -s mesos | grep Version | awk '{print $2}')
+  MIT_MESOS_LIB_DOCKER_VERSION=$(docker exec $1 sh -c "dpkg -s mesos | grep Version" |  awk '{print $2}')\
+
+  if [[ ! "$MIT_MESOS_HOST_VERSION" == "$MIT_MESOS_LIB_DOCKER_VERSION" ]]; then
+    printMsg "WARN: Host and Docker images have different versions, Host:$MIT_MESOS_HOST_VERSION, Image: $MIT_MESOS_LIB_DOCKER_VERSION (image always reflects the latest version). Pls upgrade host."
+  fi
+}
+
+
 #start master
 function start_master {
 
   dip=$(docker_ip)
-  start_master_command="/usr/sbin/mesos-master --ip=$dip $MESOS_MASTER_CONFIG"
+
+  start_master_command="/usr/sbin/mesos-master --ip=$dip $(quote_if_non_empty '$MESOS_MASTER_CONFIG')"
```

Comment:
Nope. It has to be `"`, not `'`.
Like this it generates `/usr/sbin/mesos-master --ip=192.168.0.1 $MESOS_MASTER_CONFIG`, without the substitution.

Also, the fix should be done at for each call of `quote_if_non_empty`.
It might be simple to replace `echo "\"$1\""` by `echo "\"$@\""` in `quote_if_non_empty`.